### PR TITLE
Keep all traces

### PR DESCRIPTION
**Pull request by jasonLaster**
This tells the backend to save all the honeycomb events so that we'll have an event in honeycomb for every protocol request we see in the viewer.

### CODE REVIEW

File: src/ui/actions/session.ts
Comment (jcmorrow) [inline src/ui/actions/session.ts:200]: this needs to be `sampleAllTraces`

File: packages/shared/user-data/GraphQL/config.ts
Comment (jcmorrow) [inline packages/shared/user-data/GraphQL/config.ts:83]: I think the FE team prefers that we keep the `backend_` tag here.

File: src/ui/actions/session.ts
Comment (jcmorrow) [reply src/ui/actions/session.ts:200]:
```suggestion
        sampleAllTraces: userData.get("backend_sampleAllTraces"),
```

File: packages/shared/user-data/GraphQL/config.ts
Comment (jcmorrow) [reply packages/shared/user-data/GraphQL/config.ts:83]:
```suggestion
  backend_sampleAllTraces: {
```

**Review by jcmorrow — COMMENTED** [review]
I think the suggestions batch here should set ya straight.

File: src/ui/components/ProtocolViewer/hooks/useHoneycombQueryLink.ts
Comment (jcmorrow) [inline src/ui/components/ProtocolViewer/hooks/useHoneycombQueryLink.ts:26]: I think this is probably fine, although I also think that in every case there will be exactly one result here, so sort should not matter.

File: packages/protocol/socket.ts
Comment (jcmorrow) [inline packages/protocol/socket.ts:39]: I'm not sure what's going on here, seems like somehow your prettier and trunk's prettier disagree?

**Review by jcmorrow — APPROVED** [review]
AFAICT, this code looks fine in how it interacts with Honeycomb and the backend. It seems like it has failing tests and formatting issues, and I have no idea if that's expected or not, but probably should not merge without checking with FE as well (but I leave that to your discretion).

**Comment by jasonLaster** [comment]
Thanks.

File: packages/shared/user-data/GraphQL/config.ts
Comment (bvaughn) [inline packages/shared/user-data/GraphQL/config.ts:87]: I guess it isn't _quite_ what the high-risk flag was added for but this setting should probably be flagged as high-risk (since we don't want to it left on in most cases)?

```js
  highRisk: Boolean(true),
```

File: packages/protocol/socket.ts
Comment (jasonLaster) [reply packages/protocol/socket.ts]: yeah - i think that's true.

File: packages/shared/user-data/GraphQL/config.ts
Comment (jasonLaster) [reply packages/shared/user-data/GraphQL/config.ts:87]: This is a case where i'd like to be able to leave it on for a bit and see what is happening on the backend

File: src/ui/components/ProtocolViewer/hooks/useHoneycombQueryLink.ts
Comment (jasonLaster) [reply src/ui/components/ProtocolViewer/hooks/useHoneycombQueryLink.ts:26]: yeah - not sure how what the best strategy is.

File: packages/shared/user-data/GraphQL/config.ts
Comment (bvaughn) [reply packages/shared/user-data/GraphQL/config.ts:87]: Isn't that kind of circumstantial? (Would you want more people turning this on and leaving it on?)

The run book you shared says:
> Please do not keep this setting on more than you have to. We almost always operate right at the limit of our honeycomb budget and if we keep all internal traces we will go over our Honeycomb limits and start dropping valuable data. 🙏

To me, that sounds like a reasonable use of the high-risk flag UI.

**Pull request merged**
